### ISSUE TITLE
SUM function on float amount number

### ISSUE THREAD

**Issue by ducbot88** [NLPchina/elasticsearch-sql#674]
Hi NLPChina,

I tried to sum the float amount by this query:

**SELECT SUM(amount) from customer.**

There is 2 rows with amount 0.001 and 0.002.

the SUM amount is displayed is 0.00300000001234

Please kindly to support me to validate the sum amount in this case.

Hope to hear from you soon.
Duc Hoang

**Comment by shi-yuan** [comment]
have a look at [Elasticsearch converting float numbers when aggregating](https://github.com/elastic/elasticsearch/issues/9850)

you should choose appropriate [Numeric datatypes](https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html)

**Comment by tinh1115** [comment]
@shi-yuan : which type should I use to have the correct result (0.003) to be returned if currently the field type I'm using is "float"?
is it double or something else?

**Comment by tinh1115** [comment]
@shi-yuan : I just had a quick test with below scenario:
- field1 and field2 are float
- running an aggs as:

`"aggs": {
	"sum_field_1": {
	  "sum": {
		"field": "field1"
	  }
	},
	"sum_field_2": {
	  "sum": {
		"field": "field2"
	  }
	}
}`

the result for sum_field_1 and sum_field_2 are double (0.00300000001234), it's not 0.003.
based on the answer here: https://github.com/elastic/elasticsearch/issues/9850 , I think the float value was passed to double as return as double.

In my case, I changed the field type for field1 and field2 to double and I got correct result.
So, I think the field type should be "double" to get expecting result 0.003
